### DISCUSSION TITLE
Shulker Boxes in Kits

### DISCUSSION THREAD

**Discussion by mikayden** [EssentialsX/Essentials#4466] [Q&A] [answered]
Hi, I run my own minecraft server with essentialsX because well its essential for a minecraft server. I wanted to make an ultimate kit that comes with all the ranks loot, but there just wasn't enough inventory space so I was thinking of shulker boxes. When I added the loot to the shulker boxes then made a kit when I received the kit the shulker boxes were empty. Is there any other way of adding more items than inventory space to a kit, or is there a way to allow full shulker boxes?

**Comment by triagonal** [comment, marked as answer]
This is due to server limitations preventing the `/createkit` command from capturing all of an item's data.

You have a few options to add shulker boxes with items to kits:
1. If you're using Paper 1.15.2+ and the latest stable build (2.19.0), you can use the `use-nbt-serialization-in-createkit` config option to save all of your item data. Note that this will save your kit information in a format that cannot be edited by hand - you may want to turn it on just to save the shulker box, and then turn it back off.
2. You can use a /minecraft:give command generator to create the shulker box and then add the command inside your kit like so, replacing the selector or player's name with `{USERNAME}`
```
items:
  - /minecraft:give {USERNAME} ...
```
